File: Java/프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int cnt = 0;
        long a=num;
        while(cnt<500){
            if(a==1){
                return cnt;
            }
            if(a%2==0){
                a/=2;
            }
            else{
                a=(a*3)+1;
            }
            cnt++;
        }
        return -1;
    }
}

File: Kotlin/프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.kt
class Solution {
    fun solution(num: Int): Int {
        var answer : Long = num.toLong()
        var count = 0
        
        while(count < 500 && answer!=1L) {
            if(answer%2==0L) answer = answer/2 else answer = 3*answer+1
            count++
        }
        if(count == 500) return -1
        return count
    }
}